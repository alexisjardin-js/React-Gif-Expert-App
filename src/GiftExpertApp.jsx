import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GiftExpertApp = () => {

    const [categories, setCategories] = useState([]);

    const onAddCategory = (onNewCategory) => {

        if (categories.includes(onNewCategory)) return;

        setCategories([onNewCategory, ...categories]);

    }

    return (
        <>
            <h1>GiftExpertApp</h1>

            <AddCategory
                onNewCategory={onAddCategory}
            />

            {
                categories.map((category) => (
                    <GifGrid
                        key={category}
                        category={category} />
                ))
            }

        </>
    )
}
