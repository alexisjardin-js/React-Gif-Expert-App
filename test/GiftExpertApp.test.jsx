import { render, screen } from "@testing-library/react";
import { GiftExpertApp } from "../src/GiftExpertApp";

describe('Probando <GiftExpertApp/>', () => {


   test('debe hacer match con el snapshot', () => {

       const { container }= render(<GiftExpertApp/>);
       expect(container).toMatchSnapshot();
     
   });
   
   test('', () => {
     
   })
   
  
});
