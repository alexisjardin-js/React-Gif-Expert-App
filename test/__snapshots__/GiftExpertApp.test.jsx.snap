// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GiftExpertApp/>  1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`Probando <GiftExpertApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
